app/forms/otp_verification_form.rb

```diff
@@ -28,6 +56,11 @@ def otp_code_length
     TwoFactorAuthenticatable::DIRECT_OTP_LENGTH
   end
 
+  def otp_expired?
+    return if user.direct_otp.blank?
+    Time.zone.now > user.direct_otp_sent_at + TwoFactorAuthenticatable::DIRECT_OTP_VALID_FOR_SECONDS
```

Comment:
thinking out loud, would this read better using [`#future`](https://api.rubyonrails.org/v7.0.4/classes/DateAndTime/Calculations.html#method-i-future-3F)

```suggestion
    (user.direct_otp_sent_at + TwoFactorAuthenticatable::DIRECT_OTP_VALID_FOR_SECONDS).future?
```
(reasons to avoid this would be so we could inject a direct `now` value for more deterministic tests)

Reply:
Just at a glance: Would it be the opposite (`.past?`) ?

Reply:
🤦 yes I think it would be `#past?`

Reply:
> 🤦 yes I think it would be `#past?`

Updated to `#past?` in 02fadc8fa.